libs/api/libbpf_program.cpp

```diff
@@ -434,17 +434,37 @@ bpf_prog_get_next_id(uint32_t start_id, uint32_t* next_id)
 int
 libbpf_prog_type_by_name(const char* name, enum bpf_prog_type* prog_type, enum bpf_attach_type* expected_attach_type)
 {
-    if (prog_type == nullptr || expected_attach_type == nullptr) {
+    if (name == nullptr || prog_type == nullptr || expected_attach_type == nullptr) {
         return libbpf_err(-EINVAL);
     }
 
-    ebpf_result_t result = ebpf_get_bpf_program_type_by_name(name, prog_type, expected_attach_type);
+    ebpf_program_type_t ebpf_program_type;
+    ebpf_attach_type_t ebpf_attach_type;
+    ebpf_result_t result = ebpf_get_program_type_by_name(name, &ebpf_program_type, &ebpf_attach_type);
     if (result != EBPF_SUCCESS) {
         ebpf_assert(result == EBPF_KEY_NOT_FOUND);
-        errno = ESRCH;
-        return -1;
+        return libbpf_err(-ESRCH);
     }
 
+    *prog_type = get_bpf_program_type(&ebpf_program_type);
+    *expected_attach_type = get_bpf_attach_type(&ebpf_attach_type);
+
+    return 0;
+}
+
+int
+libbpf_attach_type_by_name(const char* name, enum bpf_attach_type* attach_type)
+{
+    if (name == nullptr || attach_type == nullptr) {
+        return libbpf_err(-EINVAL);
+    }
+
+    const ebpf_attach_type_t* ebpf_attach_type = get_attach_type_windows(name);
+    if (ebpf_attach_type == &EBPF_ATTACH_TYPE_UNSPECIFIED) {
+        return libbpf_err(-ESRCH);
+    }
+
+    *attach_type = get_bpf_attach_type(ebpf_attach_type);
```

Comment:
Same as earlier, we can instead have one call to `get_bpf_program_and_attach_type()`

Reply:
I don't see any API called get_bpf_program_and_attach_type in the code base.

Reply:
Updated